pytorch_lightning/utilities/auto_restart.py

```diff
@@ -124,6 +125,15 @@ def load_state_dict(self, state_dict: Dict[int, Any], workers_initialized: bool
         self._current_iteration = state_dict[self.worker_id]["current_iteration"]
         self.restarting = True
 
+    def _get_rng_states(self):
+
+        def _collect(gen: torch.Generator):
+            return gen.get_state()
+
+        states = apply_to_collection(self.__dict__, torch.Generator, _collect)
+        states["__global"] = torch.get_rng_state()
```

Comment:
why "global"?

Reply:
the global rng state
this code is unused, we are trying to implement capturing the rng state. 
this is wip

Reply:
kk, meant to ask "why global and not rng_state"

Reply:
all the entries in this dict will be rng states from different generators, so for my intuition it does not make sense to include `rng_state` to everyone :) Same goes for the global state :)